lib/rules/array-callback-return.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * @fileoverview Rule to enforce return statements in callbacks of array's methods
+ * @author Toru Nagashima
+ * @copyright 2015 Toru Nagashima. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+var TARGET_NODE_TYPE = /^(?:Arrow)?FunctionExpression$/;
+var TARGET_METHODS = /^(?:every|fi(?:lter|nd(?:Index)?)|map|reduce(?:Right)?|so(?:me|rt))$/;
```

Comment:
I challenged: http://jsperf.com/if-str-in-candidates-indexof-vs-regexp

As expected, RegExp is the fastest.
I was surprised a use of object keys is the slowest.